elastic/kibana
# [Fleet] Allow to move agent policy to another space.

## Summary

Resolve https://github.com/elastic/kibana/issues/182733

Test are failing as it need 189387 to be merged but the PR should be ready for a first pass of review.
https://github.com/elastic/kibana/pull/189387 should be merged before (to be able to use `updateObjectsSpaces` saved object must be multiple.)




## API changes

The update agent policy API now accept a `space_ids` property, when space change, we update the policy saved object (agent policy and package policy). Then we update the related Fleet system indices enrollment tokens and agents.

We only allow to change the spaces to space where the has the `AgentPolicies:All` privileges.

Introduce a new internal API  that return the space a user has access with `AgentPolicies:All` 

```
/internal/fleet/agent_policies_spaces
```

## UI Changes 

<img width="1511" alt="Screenshot 2024-08-09 at 9 52 33 AM" src="https://github.com/user-attachments/assets/c6b9ec79-8ee5-441e-92c7-6216ce768102">


## Todo

- [x] Policy Creation

## Test

With space awareness enabled (`useSpaceAwareness` feature flag)

1. create an agent policy and add agent to that policy
2. go to settings of that policy and try to change space => verify related entities are updated too.
